runtime/tests/checker/arrays_dictionaries_test.go

```diff
@@ -753,6 +753,19 @@ func TestCheckInvalidArrayRemoveLastFromConstantSized(t *testing.T) {
 	assert.IsType(t, &sema.NotDeclaredMemberError{}, errs[0])
 }
 
+func TestCheckArrayIndexOf(t *testing.T) {
```

Comment:
Please also add some tests for invalid cases, like unequatable values, and resource-kinded values. See the test cases above

Reply:
ok, will do.

Reply:
I have tried adding more tests here, are there any in particular you feel are missing now.